### PR TITLE
Make Rtrace ESM by default, with UMD fallback

### PR DESCRIPTION
**Pull request by dcodeIO**
Does what https://github.com/AssemblyScript/assemblyscript/pull/1513 did for the loader for RTrace as well.

- [x] I've read the contributing guidelines

### CODE REVIEW

File: lib/rtrace/umd/index.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+  
+  Error.stackTraceLimit = 50;
+  
+  function trimStacktrace(stack, levels) {
+    return stack.split(/\r?\n/).slice(1 + levels);
+  }
+  
+  function isTLSF(stack) {
+    return stack[0].startsWith("    at ~lib/rt/tlsf/");
+  }
```
Comment (MaxGraey) [inline lib/rtrace/umd/index.js:33]:
```suggestion
  const RT_TLSF = "~lib/rt/tlsf/";
  // ...
  function isTLSF(stack) {
    return stack[0].startsWith(`    at ${RT_TLSF}`);
  }
```

File: lib/rtrace/umd/index.js [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+      this.allocSites = new Map();
+      this.allocCount = 0;
+      this.resizeCount = 0;
+      this.moveCount = 0;
+      this.freeCount = 0;
+      this.incrementCount = 0;
+      this.decrementCount = 0; // The following hooks cannot just be on the prototype but must be
+      // bound so the Rtrace instance can be used as a WebAssembly import.
+  
+      this.onalloc = this.onalloc.bind(this);
```
Comment (MaxGraey) [inline lib/rtrace/umd/index.js:61]: How about use arrow functions instead bindings? 
instead `this.load_val_i32 = this.load_val_i32.bind(this)` just use:
```js
load_val_i32 = (id, value) => {
   return value;
}
```
and for rest methods use the same.

Comment (dcodeIO) [reply lib/rtrace/umd/index.js:61]: Do you mean moving all of the declarations into the ctor like that?

Comment (MaxGraey) [reply lib/rtrace/umd/index.js:61]: No, just remove `this.load_val_i32 = this.load_val_i32.bind(this)` and friends and bind this via arrow function during declaration like:

```js
class RTrace {

   ....
   load_val_i32 = (id, value) => {
     return value;
  }
  ....
}
```
instead:
```js
class RTrace {
   constructor() {
       this.load_val_i32 = this.load_val_i32.bind(this);
   }
   ....
   load_val_i32(id, value) {
     return value;
  }
}
```

Comment (dcodeIO) [reply lib/rtrace/umd/index.js:61]: Hmm, how would that look for

```ts
  load_ptr(id, bytes, offset, address) {
    this.accessShadow(address + offset, bytes, true);
    return address;
  }
```

when `this` is accessed?

Comment (dcodeIO) [reply lib/rtrace/umd/index.js:61]: Also note that `bind`ing the functions makes them usable without a proper `this` context, i.e. as a Wasm import that naturally loses the `this` context. Wondering if that'd still work when not `bind`ing.

Comment (MaxGraey) [reply lib/rtrace/umd/index.js:61]: instance methods defined via arrow function grabbed current context (this of class or object). It's typical alternative for `this.method = this.method.bind(this)`

Comment (dcodeIO) [reply lib/rtrace/umd/index.js:61]: Hmm

```
SyntaxError: unknown: Support for the experimental syntax 'classProperties' isn't currently enabled (179:11):

  177 |   // Runtime instrumentation
  178 |
> 179 |   onalloc = (ptr) => {
```

Comment (MaxGraey) [reply lib/rtrace/umd/index.js:61]: Hmm, class properties should work since `V8 7.2`

Comment (dcodeIO) [reply lib/rtrace/umd/index.js:61]: Yeah, I think that's a Babel-ish error. May I suggest to make this a follow-up? Otherwise sounds good if it works!

Comment (MaxGraey) [reply lib/rtrace/umd/index.js:63]: For babel you should just add this: https://babeljs.io/docs/en/babel-plugin-syntax-class-properties. It add only syntax for parser, but don't do actually transformation (lowering)

File: lib/rtrace/umd/index.js [hunk 3]
```diff
@@ -28,8 +28,10 @@ var rtrace = (function(exports) {
     return stack.split(/\r?\n/).slice(1 + levels);
   }
   
+  const RT_TLSF = "~lib/rt/tlsf/";
+  // ...
```
Comment (MaxGraey) [inline lib/rtrace/umd/index.js:32]: Only one suggestion. Move this on top to other constans

File: lib/rtrace/umd/index.js (hunk 1 above)
Comment (a9906533) [reply lib/rtrace/umd/index.js]: > ```suggestion
> 
>   const RT_TLSF = "~lib/rt/tlsf/";
> 
>   // ...
> 
>   function isTLSF(stack) {
> 
>     return stack[0].startsWith(`    at ${RT_TLSF}`);
> 
>   }
> 
> ```

**Pull request merged**